bin/cylc-profile-battery

```diff
@@ -0,0 +1,925 @@
+#!/usr/bin/env python
+"""Orchestrates experiments to profile the performance of cylc at different
+versions."""
+import glob
+import hashlib
+import itertools
+import json
+import optparse
+import os
+import random
+import shutil
+import sys
+import tempfile
+import time
+
+import cylc.profiling as prof
+from cylc.profiling.analysis import (make_table, print_table, plot_results)
+import cylc.profiling.git as git
+
+RUN_DOC = r"""cylc profile-battery [-e [EXPERIMENT ...]] [-v [VERSION ...]]
+
+Run profiling experiments against different versions of cylc. A list of
+experiments can be specified after the -e flag, if not provided the experiment
+"complex" will be chosen. A list of versions to profile against can be
+specified after the -v flag, if not provided the current version will be used.
+
+Experiments are stored in dev/profile-experiments, user experiments can be
+stored in .profile/experiments. Experiments are specified without the file
+extension, experiments in .profile/ will be chosen before those in dev/.
+
+IMPORTANT: See dev/profile-experiments/example for an experiment template with
+further details.
+
+Versions are any valid git identifiers i.e. tags, branches, commits. To compare
+results to different cylc versions either:
+    * Supply cylc profile-battery with a complete list of the versions you wish
+      to profile, it will then provide the option to checkout the required
+      versions automatically.
+    * Checkout each version manually running cylc profile-battery against only
```

Comment:
Instead of checkout, I wonder if it is better to create a separate tree using `git archive` or similar - so that the clone can still be used while the test is ongoing.